app/services/provider_interface/save_condition_statuses.rb

```diff
@@ -0,0 +1,60 @@
+module ProviderInterface
+  class SaveConditionStatuses
+    include ImpersonationAuditHelper
+
+    def initialize(actor:, application_choice:, conditions:)
```

Comment:
Maybe the interface for this should change?

I like that we use the offer.conditions in the component to render in the UI, and we can temporarily set statuses without saving. The idea here is that we then just call `save` on the conditions to persist the change.

We do get a bullet eager loading warning with the current setup

Reply:
Updated to use `dup` to render and store data about the updated conditions. This way there's no confusing unsaved changes on the conditions